lib/services/user.go

```diff
@@ -225,6 +239,10 @@ type UserSpecV2 struct {
 	// Roles is a list of roles assigned to user
 	Roles []string `json:"roles,omitempty"`
 
+	// Context is a map[string][]string that contains additional information about
+	// the user that is used when constructing roles for a User.
+	Context map[string][]string `json:"context,omitempty"`
```

Comment:
do we _ever_ need to dump the context into JSON/YAML? maybe `json:"-"` will be better?